External/FEXCore/Source/Utils/Allocator.cpp

```diff
@@ -145,6 +145,8 @@ namespace FEXCore::Allocator {
   #define STEAL_LOG(...) // fprintf(stderr, __VA_ARGS__)
 
   std::vector<MemoryRegion> StealMemoryRegion(uintptr_t Begin, uintptr_t End) {
+    void *StackLocation = alloca(0);
```

Comment:
Is it safe to call this at the top of the function, rather than after all other variables are declared? If any of the variables are spilled, the stack location would change by the time we use `StackLocation`.

Reply:
This is safe since we are only using the stack location to find which VMA is holding our stack when parsing through `/proc/self/maps`.

Once we find the VMA that contains the stack range we know that we have the right one and allocate the prior one so our stack can keep growing.

The way this code works is a bit hard to parse since I didn't write it. I'll likely rewrite this in the future to make it less of a nightmare.